src/util/transaction_utils.rs

```diff
@@ -1,19 +1,212 @@
-use bitcoin::blockdata::transaction::TxOut;
+use bitcoin::blockdata::transaction::{TxIn, TxOut};
 
 use std::cmp::Ordering;
 
-pub fn sort_outputs<T>(outputs: &mut Vec<(TxOut, T)>) { //TODO: Make static and put it in some utils somewhere (+inputs sorting)
+pub fn sort_outputs<T>(outputs: &mut Vec<(TxOut, T)>) {
 	outputs.sort_unstable_by(|a, b| {
 		if a.0.value < b.0.value {
 			Ordering::Less
 		} else if b.0.value < a.0.value {
 			Ordering::Greater
-		} else if a.0.script_pubkey[..] < b.0.script_pubkey[..] { //TODO: ordering of scripts shouldn't be len-based
+		} else if a.0.script_pubkey[..] < b.0.script_pubkey[..] {
 			Ordering::Less
-		} else if b.0.script_pubkey[..] < a.0.script_pubkey[..] { //TODO: ordering of scripts shouldn't be len-based
+		} else if b.0.script_pubkey[..] < a.0.script_pubkey[..] {
 			Ordering::Greater
 		} else {
 			Ordering::Equal
 		}
 	});
 }
+
+pub fn sort_inputs<T>(inputs: &mut Vec<(TxIn, T)>) {
+	inputs.sort_unstable_by(|a, b| {
+		if a.0.prev_hash.into_le() < b.0.prev_hash.into_le() {
+		Ordering::Less
+		} else if b.0.prev_hash.into_le() < a.0.prev_hash.into_le() {
+			Ordering::Greater
+		} else if a.0.prev_index < b.0.prev_index {
+			Ordering::Less
+		} else if b.0.prev_index < a.0.prev_index {
+			Ordering::Greater
+		} else {
+			Ordering::Equal
+		}
+	});
+}
+
+#[cfg(test)]
+mod tests {
+	use super::*;
+
+	use bitcoin::blockdata::script::{Script, Builder};
+	use bitcoin::blockdata::transaction::TxOut;
+    use bitcoin::util::hash::Sha256dHash;
```

Comment:
There's a few tab/space things in the diff here - I always set my editor to auto-detect whether it should use tabs or spaces based on the file, and also highlight tabs and spaces differently so they're instantly visible.

Reply:
crap. thought I'd fixed all of these. My apologies.

Reply:
No big deal, I'm generally happy to clean up simple stuff like that post-merge (see https://github.com/rust-bitcoin/rust-lightning/pull/90/commits/7f4602596cf9c8d850f514d7b7ca9bc68e95e2dd)